src/consts.rs

```diff
@@ -3,6 +3,7 @@ use crate::types::{KeyLength, SubtreeHeight, SubtreeHeightKind};
 pub const PREFIX_STATE: &[u8] = &[0];
 pub const PREFIX_SMT: &[u8] = &[1];
 pub const PREFIX_DIFF: &[u8] = &[2];
+pub const PREFIX_CURRENT_STATE: &[u8] = &[3];
```

Comment:
Can all these prefixes be translated into `enum`?

Reply:
I guess there is no a good approach to change them into `enum` but I found a solution for struct implementation, I changed that and push it in separate `commit`, you can check it now.